src/test/java/net/sf/jabref/exporter/BibtexDatabaseWriterTest.java

```
@@ -40,186 +37,175 @@
-        stringWriter = new StringWriter();
+        // Write to a string instead of to a file
+        databaseWriter = new BibtexDatabaseWriter<>(StringSaveSession::new);
+
         database = new BibDatabase();
         metaData = new MetaData();
         bibtexContext = new BibDatabaseContext(database, metaData, new Defaults(BibDatabaseMode.BIBTEX));
     }
 
     @Test(expected = NullPointerException.class)
-    public void writeWithNullWriterThrowsException() throws IOException {
-        databaseWriter.writePartOfDatabase(null, bibtexContext, Collections.emptyList(), mock(SavePreferences.class));
-    }
-
-    @Test(expected = NullPointerException.class)
-    public void writeWithNullContextThrowsException() throws IOException {
-        databaseWriter.writePartOfDatabase(mock(Writer.class), null, Collections.emptyList(), new SavePreferences());
+    public void writeWithNullContextThrowsException() throws Exception {
+        databaseWriter.savePartOfDatabase(null, Collections.emptyList(), new SavePreferences());
     }
 
     @Test(expected = NullPointerException.class)
-    public void writeWithNullEntriesThrowsException() throws IOException {
-        databaseWriter.writePartOfDatabase(mock(Writer.class), bibtexContext, null, new SavePreferences());
+    public void writeWithNullEntriesThrowsException() throws Exception {
+        databaseWriter.savePartOfDatabase(bibtexContext, null, new SavePreferences());
     }
 
     @Test(expected = NullPointerException.class)
-    public void writeWithNullPreferencesThrowsException() throws IOException {
-        databaseWriter.writePartOfDatabase(mock(Writer.class), bibtexContext, Collections.emptyList(), null);
+    public void writeWithNullPreferencesThrowsException() throws Exception {
+        databaseWriter.savePartOfDatabase(bibtexContext, Collections.emptyList(), null);
     }
 
     @Test
-    public void writeEncoding() throws IOException {
+    public void writeEncoding() throws Exception {
```

Why did you remove all specific Exceptions?
Better let it throw the most specific exception. Otherwise stacktrace can be hard to find.